var/spack/repos/builtin/packages/py-ctranslate2/package.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyCtranslate2(CMakePackage, PythonExtension, CudaPackage):
+    """CTranslate2 is a C++ and Python library for efficient inference with
+    Transformer models."""
+
+    homepage = "https://github.com/OpenNMT/CTranslate2"
+    git = "https://github.com/OpenNMT/CTranslate2.git"
+
+    maintainers("meyersbs", "aweits")
+
+    version("3.12.0", tag="v3.12.0", submodules=True)
+
+    extends("python")
+
+    # From setup.py, CMakeLists.txt, requirements.txt
+    depends_on("py-setuptools", type="build")
+    depends_on("cmake@3.15:", type="build")
+    depends_on("py-pip", type="build")
+    depends_on("py-pybind11", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-pyyaml@5.3:6", type=("build", "run"))
+
+    # From build errors
+    depends_on("py-intel-openmp", type=("build", "run"))
+    # From https://opennmt.net/CTranslate2/installation.html#build-options
+    variant("mkl", default=True, description="Build with mkl")
+    variant("cuda", default=False, description="Build with cuda")
+    variant("cudnn", default=False, description="Build with cudnn")
+    depends_on("intel-mkl@2019.5:", when="+mkl", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("mkl@2019.5:", when="+mkl", type=("build", "run"))
```
The version specifier prob won't work with the virtual dep though

Reply:
Looks like this change does work